content/hardware/07.opta/opta-family/opta/tutorials/17.opta-mqtt/content.md

```diff
@@ -0,0 +1,555 @@
+    mqttClient.beginMessage(MQTT_TOPIC);
+    // Add message content (in this case, "hello")
+    mqttClient.print("hello");
+    // End the MQTT message and send it to the broker
+    mqttClient.endMessage();
+    previous_time = current_time;  // Update previous time to current time
+  }
+}
+```
+
+The complete code might look like this:
+
+```cpp
+#include <ArduinoMqttClient.h>  // Include the ArduinoMqttClient library to handle MQTT connection
+#include <WiFi.h>               // Include the WiFi library to handle WiFi connection
+
+#include "arduino_secrets.h"   // Include the arduino_secrets.h file containing WiFi and MQTT connection credentials
+
+WiFiClient wifiClient;              // Create a wifiClient object of type WiFiClient to handle WiFi connection
+MqttClient mqttClient(wifiClient);  // Create an mqttClient object of type MqttClient using wifiClient to handle MQTT connection
+
+unsigned long current_time, previous_time = millis();  // Variables for time control
+
+void setup() {
+  // Start WiFi connection using provided credentials
+  while (WiFi.begin(SECRET_SSID, SECRET_PASS) != WL_CONNECTED) {  // Keep attempting connection until connected
+    Serial.print(".");                                            // Print a dot on serial to indicate connection attempt
+    delay(5000);                                                  // Wait 5 seconds between connection attempts
+  }
+
+  mqttClient.setUsernamePassword(MQTT_USER, MQTT_PASSWORD);  // Set the username and password for authentication with the MQTT broker
+
+  // Connect to MQTT broker using provided credentials
+  if (!mqttClient.connect(MQTT_BROKER, MQTT_PORT)) {  // If MQTT connection fails
+    while (1)
+      ;  // Halt program execution in an infinite loop
+  }
+
+  previous_time = millis();  // Set previous time to current time
+}
+
+void loop() {
+  mqttClient.poll();  // Check for incoming messages from MQTT broker
+  current_time = millis();                      // Get current time in milliseconds
+  if (current_time - previous_time >= 10000) {  // If 10 seconds have elapsed
+    previous_time = current_time; //Update previous_time with current_time
+    // Begin a new MQTT message on the specified topic
+    mqttClient.beginMessage(MQTT_TOPIC);
+    // Add message content (in this case, "hello")
+    mqttClient.print("hello");
+    // End the MQTT message and send it to the broker
+    mqttClient.endMessage();
+    previous_time = current_time;  // Update previous time to current time
+  }
+}
+```
+
+This code will produce the same transmission effect already observed previously but with the advantage of keeping the **loop()** function free for other operations and avoiding interference with the synchronization necessary for connecting to the MQTT cloud.
+
+Of course, the "hello" message was used for example purposes; you can customize this message by sending the value of a sensor or other information based on your needs.
```

Comment:
```suggestion
The "hello" message was used for example purposes; you can customize this message by sending the value of a sensor or other information based on your needs.
```

Reply:
I suggest to avoid phrasing such as "of course". We don't want to make the assumption that this is obvious to everyone. The sentence works just as well with this removed.